File: tests/leetcode/x/scala/0014.scala
import scala.io.Source

object Main extends App {
  val tokens = Source.stdin.mkString.split("\\s+").filter(_.nonEmpty)
  if (tokens.nonEmpty) {
    var idx = 0
    val t = tokens(idx).toInt
    idx += 1
    def lcp(strs: Array[String]): String = {
      var prefix = strs(0)
      while (!strs.forall(_.startsWith(prefix))) prefix = prefix.dropRight(1)
      prefix
    }
    val out = collection.mutable.ArrayBuffer[String]()
    for (_ <- 0 until t) {
      val n = tokens(idx).toInt; idx += 1
      val strs = tokens.slice(idx, idx + n); idx += n
      out += s""""${lcp(strs)}""""
    }
    print(out.mkString("\n"))
  }
}

File: tests/leetcode/x/clojure/0014.clj
(ns main)

(defn lcp [strs]
  (loop [prefix (first strs)]
    (if (every? #(.startsWith ^String % prefix) strs)
      prefix
      (recur (subs prefix 0 (dec (count prefix)))))))

(let [tokens (vec (filter seq (clojure.string/split (slurp *in*) #"\s+")))]
  (when (seq tokens)
    (loop [idx 1
           t (Integer/parseInt (tokens 0))]
      (when (> t 0)
        (let [n (Integer/parseInt (tokens idx))
              strs (subvec tokens (inc idx) (+ (inc idx) n))]
          (println (str "\"" (lcp strs) "\""))
          (recur (+ idx 1 n) (dec t)))))))
